src/main/java/report_use_cases/ReportInteractor.java

```diff
@@ -0,0 +1,59 @@
+package report_use_cases;
+
+import entities.Report;
+import entities.ReportFactory;
+import entities.Review;
+import entities.User;
+import report_screens.ManipulateReviewDs;
+import report_screens.ReportCreationFailure;
+import java.time.LocalDateTime;
+
+public class ReportInteractor implements reportInputBoundary {
+
+    final reportDsGateway reportDsGateway;
+
+    final ReportFactory reportFactory;
+
+    final Excalibur excalibur;
+
+
+    public ReportInteractor(reportDsGateway reportDsGateway, ReportFactory reportFactory, Excalibur excalibur) {
+        this.reportDsGateway = reportDsGateway;
+        this.excalibur = excalibur;
+        this.reportFactory = reportFactory;
+    }
+
+    @Override
+    public void Create(ReportRequestModel reportRequestModel){
+
+        //if report already exists
+        if(reportDsGateway.existsReportByReporterAndReview(reportRequestModel.getReporter().getUsername(), reportRequestModel.getReview().getID())){
+            throw new ReportCreationFailure("Report failed: You've already reported this review");
+
+            // check if the reporter is banned
+        } else if (reportRequestModel.getReporter().isBanned()) {
+            throw new ReportCreationFailure("Report failed: Your account is under investigation due to massive reports");
+        }
+
+        Report report = reportFactory.create(reportRequestModel.getReason(), reportRequestModel.getReview(), reportRequestModel.getReporter().getUsername());
+        LocalDateTime now = LocalDateTime.now();
+        ReportDsRequestModel reportDsRequestModel =new ReportDsRequestModel(report.getReason(), report.getReviewContent(), report.getReview_id(), report.getReporter_username(), now.toString());
+        reportDsGateway.save(reportDsRequestModel);
+
+        //add report to targeted user
+        //reportRequestModel.getReview().getUser() IT RETURNS AN USERNAME, USE THE USERNAME TO FIND AND INITIALIZE THE USER FROM THE DATA BASE
+        //THEN .addReport()
+
+        //add report to targeted review
+        reportRequestModel.getReview().addReport();
+
+        //save the changes to the targeted user and review
+        Review updated_revivew = excalibur.execute_review();
+        User updated_user = excalibur.execute_user();
+
+        ManipulateReviewDs manipulateReviewDs = new ManipulateReviewDs();
+        manipulateReviewDs.updateReview(updated_revivew);
+
+
+    }
+}
```

Comment:
Not sure if I'm misunderstanding your process, but theres no presenter to show the success view that I can see
Let me know if it is implemented elsewhere

Reply:
yeah no presenter is being used; all messages (success OR failure view with error message) are shown to user by a popout screen, JOptionPane.showMessageDialog, within the ReportScreen class

Reply:
I can't reply to @prajogt other comment so I'll just put it here: I was going to prevent banned users from doing anything at the screen layer, but we can check in both layers to make sure

Reply:
Ah ok fair enough, make sure you change it in the CRC later then

Reply:
Will do

Reply:
Jarod i think i miss clicked something i dismissed ur approval; can u do it again? @JarodGratton